app/controllers/ArrivalsController.scala

```diff
@@ -16,37 +16,54 @@
 
 package controllers
 
+import java.net.URLEncoder
+
 import connectors.MessageConnector
 import controllers.actions.AuthAction
 import javax.inject.Inject
 import models.request.ArrivalNotificationXSD
 import play.api.mvc.{Action, ControllerComponents}
 import services.XmlValidationService
-import uk.gov.hmrc.auth.core.InsufficientEnrolments
-import uk.gov.hmrc.http.Upstream5xxResponse
 import uk.gov.hmrc.play.bootstrap.controller.BackendController
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.xml.NodeSeq
 
-
 class ArrivalsController @Inject()(cc: ControllerComponents,
                                    authAction: AuthAction,
                                    messageConnector: MessageConnector,
                                    xmlValidationService: XmlValidationService)(implicit ec: ExecutionContext) extends BackendController(cc) {
+
+  private val arrivalsFormat = """^/arrivals/(.*)$""".r
```

Comment:
Could be that the Location received from the BE service has parameters which we don't want to pass on to the 3rd party.  e.g. /movements/arrivals/123?status=success
That _should_ never happen but lets program defensively and terminate the arrival id at a ? character